### ISSUE TITLE
fluent bit pods crashing with OOM 

### ISSUE THREAD

**Issue by gauravkcldcvr** [fluent/fluent-bit#4609]
## Bug Report

Fluent bit pod randomly restarting with OOM as termination status code 137

This has been happening a lot frequently, to understand the reason I have given enough resource allocation (3Gi) and monitored with grafana, and found it gets OOM killed without even crossing the requested memory. Tried changing the parameter **Mem_Buf_Limit** from 50M to 100M but that didn't help either.  I could not find anything helpful in the pods logs so I'm not sure what causing this, also the image its using does not have a shell to login to pods.

I need some understanding what could be the cause of these memory leaks and these restarts? Have I missed anything in the configuration?

<img width="1376" alt="Screenshot 2022-01-13 at 3 30 56 PM" src="https://user-images.githubusercontent.com/93526209/149308959-db3e51d4-edab-4f32-ba67-4534f53cd4df.png">

**Environment**
* Image tag: fluent-bit:1.8.7
* Environment name and version:  Amazon EKS (1.21.2)

**Multiple restarts observed**
```
> k get pods -n logging
NAME                          READY   STATUS    RESTARTS   AGE
loki-stack-0                  1/1     Running   0          44h
loki-stack-fluent-bit-6jgvc   1/1     Running   33         4h55m
loki-stack-fluent-bit-7zwnl   1/1     Running   14         4h55m
loki-stack-fluent-bit-8jr4h   1/1     Running   29         4h56m
loki-stack-fluent-bit-xvjzf   1/1     Running   11         4h55m
loki-stack-fluent-bit-xw7fz   1/1     Running   6          4h56m
```

**Fluent bit resources allocation**

```
resources:
    limits:
      # cpu: 100m
      memory: 3Gi
    requests:
      # cpu: 100m
      memory: 2Gi
```

**Configuration** 

```
config:
    service: |
      [SERVICE]
          Daemon Off
          Flush 1
          Log_Level {{ .Values.logLevel }}
          Parsers_File parsers.conf
          Parsers_File custom_parsers.conf
          HTTP_Server On
          HTTP_Listen 0.0.0.0
          HTTP_Port {{ .Values.service.port }}
          Health_Check On
          storage.path /tmp/flb-log
          storage.backlog.mem_limit 50M
          storage.max_chunks_up 128

    ## https://docs.fluentbit.io/manual/pipeline/inputs
    inputs: |
      [INPUT]
          Name tail
          Path /var/log/containers/*.log
          # multiline.parser docker, cri
          parser docker_no_time
          Tag kube.*
          Mem_Buf_Limit 100MB
          Skip_Long_Lines On
          storage.type filesystem

    filters: |
      [FILTER]
          Name kubernetes
          Match kube.*
          Merge_Log On
          Keep_Log Off
          K8S-Logging.Parser On
          K8S-Logging.Exclude On
          Labels Off
          Annotations Off
```

**Comment by brucelee138** [comment]
@gauravkcldcvr I'm facing this issue too! Were you able to resolve this?

**Comment by lecaros** [comment]
Hi @gauravkcldcvr,
* What other symptoms do you see?
* can you share the log file?
* is that your full config?
* does this happen with the latest version? (1.8.15)

**Comment by gauravkcldcvr** [comment]
Closing this as this was resolved in using latest helm chart (1.9.3)
